frontend/src/components/ProfileContent.tsx

```diff
@@ -50,46 +61,107 @@ const useStyles = makeStyles({
+    const [age, setAge] = React.useState<number | string>()
+    const [gender, setGender] = React.useState<string>()
+    const [occupations, setOccupations] = React.useState<string[]>()
+    const [location, setLocation] = React.useState<string>()
+    const [interests, setInterests] = React.useState<string[]>()
+    const [religion, setReligion] = React.useState<string>()
+    const [editedProfileFields, setEditableProfileFields] = React.useState<ProfileFieldList | null>(null)
+
+    const readOnlyContent = 'readonly';
+    const editableContent = 'editable';
 
     // Add user profile values to list.
-    var profileFields: ProfileField[] = [
-        {
+    var profileFields: ProfileFieldList = {
+        name: {
             title: 'Full Name',
-            value: props.name
+            value: props.name,
         },
-        {
+        username: {
             title: 'Username',
-            value: props.username
+            value: props.username,
         },
-        {
+        age: {
             title: 'Age',
-            value: props.age.toString()
+            value: props.age.toString(),
         },
-        {
+        gender: {
             title: 'Gender',
-            value: props.gender
+            value: props.gender,
         },
-        {
+        occupations: {
             title: 'Occupations',
-            value: props.occupations.join(', ')
+            value: props.occupations.join(', '),
         },
-        {
+        location: {
             title: 'Location',
-            value: props.location
+            value: props.location,
         },
-        {
+        interests: {
             title: 'Interests',
-            value: props.interests.join(', ')
+            value: props.interests.join(', '),
         },
-        {
+        religion: {
             title: 'Religion',
-            value: props.religion
+            value: props.religion,
         },
-    ];
+    };
+
+    const handleEditBtnClick = (e) => {
+        e.preventDefault()
+        setEditableProfileFields(profileFields)
+        toggleEditMode(true)
+    }
+
+    const handleCancelBtnClick = (e) => {
+        e.preventDefault()
+        setEditableProfileFields(profileFields)
+        toggleEditMode(false)
+    }
+
+    const saveFields = (e) => {
+        console.log('saving fields:')
+        console.log(e.target)
+        toggleEditMode(false)
+
+        Object.keys(profileFields).forEach(key => {
+            let originalField: ProfileField = profileFields[key]
+            let editedField: ProfileField = editedProfileFields[key]
+            const { title, value } = editedField
+
+            // Test if field was changed.
+            if (originalField.value != value) {
+                console.log(`${title} was changed to: ${value}`)
+                originalField.value = value
+            }
+            else {
+                console.log(`${title} was not changed`)
+            }
```

Comment:
There's a better way to do this, and I'm sure it exists somewhere in https://ramdajs.com/docs/, but I can't come up with it at the moment.

Reply:
yes, this is still a draft. I havent decided how it should work yet